launch/launcher.go

```diff
@@ -106,7 +106,18 @@ func (l *Launcher) env() error {
 			if err := l.Env.AddEnvDir(filepath.Join(path, "env")); err != nil {
 				return err
 			}
-			return l.Env.AddEnvDir(filepath.Join(path, "env.launch"))
+			if process.Type != "" {
+				if err := l.Env.AddEnvDir(filepath.Join(path, "env", process.Type)); err != nil {
+					return err
+				}
+			}
```

Comment:
Based on my reading of the RFC, I thought we were only accepting `env.launch/<process>/` and not accepting `env/<process>`

Reply:
I don't see an issue with removing `env/<process>`. cc @jkutner is that your understanding as well? If so, I'll move things around.

Reply:
As far as implementing `profile` scripts, I'll wait for that same question to be answered and follow the same decision. This PR will aim to match that particular spec PR.

Reply:
I actually went ahead and made the change specific to `env.launch/<process>`. I think the RFC is a bit ambiguous but having `env/<process>` would only add confusion without making the feature much easier to use.